### PR TITLE
Add routes for model detail page tabs

### PR DESCRIPTION
**Pull request by kulyk**
Epic #27581

Adds dedicated route per each model detail page's tab:

* Used by — `/model/:slug/detail/usage`
* Schema — `/model/:slug/detail/schema`
* Actions — `/model/:slug/detail/actions`

### To Verify

1. Open  `/model/:id/detail` (ensure tab isn't specified)
2. Ensure you see the "Used by" tab by default
3. Click through tabs, ensure you can see the expected content and that URLs match the ones listed above
4. Create a model based on data from a database that doesn't have actions enabled (can be turned off at admin db page) or that doesn't support actions at all (any database except Postgres and MySQL)
5. Try to open `/model/:id/detail/actions` for this model and ensure you're navigated to `/usage` instead
6. Try to open a tab that doesn't exist, e.g. `/model/:id/detail/foo-bar` and ensure you're navigated to `/usage`

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/27959)

### CODE REVIEW

File: frontend/src/metabase/models/components/ModelDetailPage/ModelDetailPage.tsx [hunk 1]
```diff
@@ -55,12 +52,23 @@ function ModelDetailPage({
...
         />
-        <TabContent value={tab} onChange={setTab}>
-          <TabList
-            value={tab}
-            options={tabs}
-            onChange={tab => setTab(tab as ModelTab)}
-          />
+        <TabContent value={tab}>
+          <TabList>
+            <Tab.Link
```
Comment (ranquild) [inline frontend/src/metabase/models/components/ModelDetailPage/ModelDetailPage.tsx:57]: We use this `Component.Subcomponent` pattern only for styled-components that are used within the root component. Can we just use `TabLink` here, so it doesn't introduce another pattern?

File: frontend/src/metabase/routes.jsx [hunk 2]
```diff
@@ -241,7 +241,10 @@ export const getRoutes = store => (
           <Route path=":slug/:objectId" component={QueryBuilder} />
         </Route>
 
-        <Route path="/model/:slug/detail" component={ModelDetailPage} />
+        <Route path="/model/:slug/detail">
+          <IndexRoute component={ModelDetailPage} />
+          <Route path=":tab" component={ModelDetailPage} />
```
Comment (ranquild) [inline frontend/src/metabase/routes.jsx:246]: It seems that we usually try not to use a generic route pattern when it's actually static. Can we refactor this to static routes with corresponding components, like:
```
<Route path="usage" component={ModelUsagePage} />
<Route path="actions" component={ModelActionsPage} />
...
```
?

This way you won't have to deal with "unknown" tabs and checking whether actions are enabled would be straightforward. WDYT?

Comment (kulyk) [reply frontend/src/metabase/routes.jsx:246]: I tried this yesterday but ran into something that made me switch to this 🤔 Let me try again just to make sure tho

File: frontend/src/metabase/core/components/TabLink/TabLink.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React, { MouseEvent, useCallback, useContext } from "react";
+import { LinkProps } from "metabase/core/components/Link";
+import { getTabId, getTabPanelId, TabContext } from "../Tab";
+import { TabLinkRoot, TabLabel } from "./TabLink.styled";
+
+export interface TabLinkProps<T> extends LinkProps {
+  value?: T;
```
Comment (WiNloSt) [inline frontend/src/metabase/core/components/TabLink/TabLink.tsx:7]: I'm not sure what are you trying to achieve with generic, but from the usage in this PR, seems like it is always string, so I wonder if this could be just `string` so it's simple?

File: frontend/src/metabase/core/components/TabLink/TabLink.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+function TabLink<T>({ value, children, onClick, ...props }: TabLinkProps<T>) {
+  const { value: selectedValue, idPrefix, onChange } = useContext(TabContext);
+  const tabId = getTabId(idPrefix, value);
+  const panelId = getTabPanelId(idPrefix, value);
+  const isSelected = value === selectedValue;
+
+  const handleClick = useCallback(
+    (event: MouseEvent<HTMLAnchorElement>) => {
+      onClick?.(event);
+      onChange?.(value);
```
Comment (WiNloSt) [inline frontend/src/metabase/core/components/TabLink/TabLink.tsx:19]: I saw `onChange` is defined here:
https://github.com/metabase/metabase/blob/f879b7f0b4909fe02fd17a403916bd14b9a72074/frontend/src/metabase/core/components/Tab/TabContext.tsx#L7

But I could never find where `onChange` is being passed to the context. I suspect now we never receive `onChange` at all. If that's the case then we could change the context signature and remove this call.

File: frontend/src/metabase/routes.jsx (hunk 2 above)
Comment (kulyk) [reply frontend/src/metabase/routes.jsx]: Nvm, whatever the problem was it's no longer relevant, switched to explicit tab definitions 👍

**Review by ranquild — APPROVED** [review]
@kulyk could the tab type be changed to a enum?

File: frontend/src/metabase/routes.jsx [hunk 5]
```diff
@@ -241,7 +241,13 @@ export const getRoutes = store => (
           <Route path=":slug/:objectId" component={QueryBuilder} />
         </Route>
 
-        <Route path="/model/:slug/detail" component={ModelDetailPage} />
+        <Route path="/model/:slug/detail">
```
Comment (WiNloSt) [inline frontend/src/metabase/routes.jsx:244]: Just realize `react-router` v3 doesn't even have `<Switch />` wonder if we should considering upgrade this some time soon to have a better and more modern API available?

File: frontend/src/metabase/routes.jsx [hunk 6]
```diff
@@ -241,7 +241,13 @@ export const getRoutes = store => (
...
+          <IndexRoute component={ModelDetailPage} />
```
Comment (WiNloSt) [inline frontend/src/metabase/routes.jsx:245]: It's possible to enter the model detail without having any tab selected. I don't think that's intended, since on stats if you go to the model detail page the **Used by** tab will be open by default. Maybe it's use `<IndexRedirect />` instead?

```suggestion
          <IndexRedirect to="usage" />
```

File: frontend/src/metabase/core/components/TabLink/TabLink.tsx (hunk 3 above)
Comment (kulyk) [reply frontend/src/metabase/core/components/TabLink/TabLink.tsx:7]: I'm just mirroring the interface of [Tab](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/core/components/Tab/Tab.tsx) from the same group of components. `TabLink` is meant to be an alternative to `Tab`; think the more similar API they have the better. WDYT?

File: frontend/src/metabase/routes.jsx (hunk 5 above)
Comment (kulyk) [reply frontend/src/metabase/routes.jsx:244]: For sure, we'd need to investigate how to better approach this first

File: frontend/src/metabase/core/components/TabLink/TabLink.tsx (hunk 3 above)
Comment (WiNloSt) [reply frontend/src/metabase/core/components/TabLink/TabLink.tsx:7]: Agree that the more similar API the better. But I still think generic here doesn't convey much. Think about using tabs, do we actually want to be able to pass anything as a value like objects, lists, maps? It'd make more sense to say maybe this can either be a `number` or a `string` ([I found the number being used here.](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/core/components/TabList/TabList.stories.tsx#L26)).

Comment (kulyk) [reply frontend/src/metabase/core/components/TabLink/TabLink.tsx:7]: Sounds good, I'd better do it in another PR though not to blow up the scope of this one too muh
